### PR TITLE
docs: update swaggo/swag v1.8.9

### PR DESCRIPTION
**Pull request by mtojek**
Related: https://github.com/coder/coder/pull/5538#discussion_r1062519526

### CODE REVIEW

File: docs/api/agents.md [hunk 1]
```diff
@@ -426,7 +426,7 @@ curl -X POST http://coder-server:8080/api/v2/workspaceagents/me/report-stats \
 
 ```json
 {
-  "report_interval": 0
+  "report_interval": -9223372036854776000
```
Comment (mtojek) [inline docs/api/agents.md:429]: For some reason the default value for integer is -Inf :)

File: docs/api/general.md [hunk 2]
```diff
@@ -310,17 +310,19 @@ curl -X GET http://coder-server:8080/api/v2/config/deployment \
...
-      "client_id": "string",
-      "id": "string",
-      "no_refresh": true,
-      "regex": "string",
-      "scopes": ["string"],
-      "token_url": "string",
-      "type": "string",
-      "validate_url": "string"
-    }
+    "value": [
```
Comment (mtojek) [inline docs/api/general.md:313]: It looks like it handles "generics" better as this should be an array.

File: docs/api/general.md [hunk 3]
```diff
@@ -410,7 +412,7 @@ curl -X GET http://coder-server:8080/api/v2/config/deployment \
         "secret": true,
         "shorthand": "string",
         "usage": "string",
-        "value": "string"
+        "value": ["string"]
```
Comment (mtojek) [inline docs/api/general.md:415]: Another generic, now correct 👍

File: docs/api/schemas.md [hunk 4]
```diff
@@ -286,10 +308,10 @@ AuthorizationCheck is used to check if the currently authenticated user (or the
...
+| `object` | [codersdk.AuthorizationObject](#codersdkauthorizationobject) | false    |              | Object can represent a "set" of objects, such as: - All workspaces in an organization - All workspaces owned by me - All workspaces across the entire product When defining an object, use the most specific language when possible to produce the smallest set. Meaning to set as many fields on 'Object' as you can. Example, if you want to check if you can update all workspaces owned by 'me', try to also add an 'OrganizationID' to the settings. Omitting the 'OrganizationID' could produce the incorrect value, as workspaces have both `user` and `organization` owners. |
```
Comment (mtojek) [inline docs/api/schemas.md:314]: A different source of description?

File: docs/api/schemas.md [hunk 5]
```diff
@@ -3416,14 +3649,35 @@ Parameter represents a set value for the scope.
...
+
+### Properties
+
+| Name          | Type   | Required | Restrictions | Description |
+| ------------- | ------ | -------- | ------------ | ----------- |
+| `*anonymous*` | string | false    |              |             |
+
+#### Enumerated Values
+
+| Property      | Value          |
```
Comment (mtojek) [inline docs/api/schemas.md:3666]: This table might be confusing for users.

File: docs/api/schemas.md [hunk 6]
```diff
@@ -4035,3 +4377,54 @@ It corresponds to the legacy derpN.tailscale.com hostnames used by older clients
...
+
+### Properties
+
+| Name          | Type    | Required | Restrictions | Description |
+| ------------- | ------- | -------- | ------------ | ----------- |
+| `*anonymous*` | integer | false    |              |             |
+
+#### Enumerated Values
+
+| Property      | Value                  |
```
Comment (mtojek) [inline docs/api/schemas.md:4395]: A lot of noise here...

**Review by mtojek — COMMENTED** [review]
@mafredri Here is the doc output if we decided to use the latest revision of swag. Please let me know your thoughts.

File: docs/api/parameters.md [hunk 7]
```diff
@@ -59,17 +59,17 @@ curl -X GET http://coder-server:8080/api/v2/parameters/{scope}/{id} \
...
+| `[array item]`         | array                                 | false    |              | [Parameter represents a set value for the scope.] |
+| `» created_at`         | string(date-time)                     | false    |              |                                                   |
+| `» destination_scheme` | `codersdk.ParameterDestinationScheme` | false    |              |                                                   |
+| `» id`                 | string(uuid)                          | false    |              |                                                   |
+| `» name`               | string                                | false    |              |                                                   |
+| `» scope`              | `codersdk.ParameterScope`             | false    |              |                                                   |
+| `» scope_id`           | string(uuid)                          | false    |              |                                                   |
+| `» source_scheme`      | `codersdk.ParameterSourceScheme`      | false    |              |                                                   |
```
Comment (mafredri) [inline docs/api/parameters.md:71]: Thinking out loud. Should we link to the types documentation here, i.e. make it clickable?

File: docs/api/schemas.md [hunk 8]
```diff
@@ -80,15 +80,37 @@
...
+
+#### Enumerated Values
+
+| Property      | Value    |
+| ------------- | -------- |
+| `*anonymous*` | `create` |
+| `*anonymous*` | `write`  |
+| `*anonymous*` | `delete` |
+| `*anonymous*` | `start`  |
+| `*anonymous*` | `stop`   |
```
Comment (mafredri) [inline docs/api/schemas.md:113]: The name/property columns don't really make sense here 🤔. We might want to rewrite how enums are represented.

File: docs/api/schemas.md (hunk 4 above)
Comment (mafredri) [reply docs/api/schemas.md:314]: I think it's the same source, but parsing might have changed or been broken? Perhaps swagger expects certain type of formatting for lists?

File: docs/api/schemas.md (hunk 5 above)
Comment (mafredri) [reply docs/api/schemas.md:3666]: Agreed, commented on another one before I saw this.

File: docs/api/schemas.md (hunk 6 above)
Comment (mafredri) [reply docs/api/schemas.md:4395]: Oh, I see what's going on. It's interpreting `time.Duration` as an enum. Maybe we update `.swaggo` with `replace time.Duration int64`

File: docs/api/templates.md [hunk 9]
```diff
@@ -175,10 +175,17 @@ Status Code **200**
...
-| `» provisioner`                      | string                            | false    |              |                                            |
+| `» provisioner`                      | `codersdk.ProvisionerType`        | false    |              |                                            |
 | `» updated_at`                       | string(date-time)                 | false    |              |                                            |
 | `» workspace_owner_count`            | integer                           | false    |              |                                            |
 
+#### Enumerated Values
+
+| Property      | Value       |
+| ------------- | ----------- |
+| `provisioner` | `echo`      |
```
Comment (mafredri) [inline docs/api/templates.md:186]: Let's not expose the `echo` provider, it's only used by tests. Can we remove it from the enum or add a replace to swaggo to just show it as a string?

**Review by mafredri — COMMENTED** [review]
Nice changes all-in-all. Added some comments but I think we'd get pretty far with a simple replace on `time.Duration`.

File: docs/api/agents.md (hunk 1 above)
Comment (mtojek) [reply docs/api/agents.md:429]: I believe we can fix it by adding the `example` annotation. Unfortunately, we need to add it to every `int` field?

Comment (mafredri) [reply docs/api/agents.md:429]: I don't think it's needed for every int, it only seems to apply to those that are `time.Duration`.

Comment (mtojek) [reply docs/api/agents.md]: You're right. Adding a record to `.swaggo` fixed the `time.Duration` issue. Thanks!

File: docs/api/schemas.md (hunk 6 above)
Comment (mtojek) [reply docs/api/schemas.md]: Fixed

File: docs/api/templates.md (hunk 9 above)
Comment (mtojek) [reply docs/api/templates.md]: Fixed. It shows only `terraform` now.

File: docs/api/schemas.md (hunk 8 above)
Comment (mtojek) [reply docs/api/schemas.md]: I refactored them a bit. Let me know your thoughts.

File: docs/api/parameters.md (hunk 7 above)
Comment (mtojek) [reply docs/api/parameters.md]: Good idea, fixed.

**Review by mafredri — APPROVED** [review]
This turned out great, nice improvements!

File: docs/api/schemas.md (hunk 8 above)
Comment (mafredri) [reply docs/api/schemas.md]: It's nice! (Optionally) we could show the base type in the enum description. But it's kind of inferable via the `json` field so not sure it's necessary.

**Pull request merged**